common.mk

```
@@ -1,7 +1,7 @@
 ARCHTUPLE=arm-none-eabi-
 DEVICE=VEX EDR V5
 
-MFLAGS=-mcpu=cortex-a9 -mfpu=neon-fp16 -mfloat-abi=hard -Os -g
+MFLAGS=-mcpu=cortex-a9 -mfpu=neon-fp16 -mfloat-abi=hard -Os -g -mthumb-interwork -mthumb
```

```suggestion
MFLAGS=-mcpu=cortex-a9 -mfpu=neon-fp16 -mfloat-abi=hard -Os -g -mthumb
```

GCC docs explicitly state that it is not necessary to use `-mthumb-interwork` ([source](https://gcc.gnu.org/onlinedocs/gcc/ARM-Options.html#index-mthumb-interwork))

> Generate code that supports calling between the ARM and Thumb instruction sets. Without this option, **on pre-v5 architectures**, the two instruction sets cannot be reliably used inside one program.

The cortex A9 uses the arm v7 architecture, so, according to these docs, there is no reason to use `-mthumb-interwork`. Not using `-mthumb-interwork` will decrease the binary size.

---

Not necessarily, the docs don't say that on post-v5 architectures it will always work without `-mthumb-interwork`, just that on pre-v5 architectures it will always fail. It depends on how the SDK was compiled, although I guess we could just test if removing the flag works.

---

Trying this in https://github.com/purduesigbots/pros/pull/708/commits/ac69470a22436d2ed3b4d7ea576035fd902f3ee7.